clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -546,6 +619,22 @@ public ConsumerRecords<K, V> poll(long timeout) {
         throw new KafkaException("method not implemented");
     }
 
+    private void maybeWakeup() {
+        if (this.closed)
+            throw new IllegalStateException("This consumer has already been closed.");
```

Comment:
Original consumer doesn't do this. Is there a good reason to change the behavior?

Reply:
there isn't and we don't need it. I'll remove this part.